test/test_webhook.rb

```diff
@@ -1,59 +1,144 @@
+
+class WebhookTest < Test::Unit::TestCase
   include Rack::Test::Methods
   include Thumbs
 
   def app
     ThumbsWeb
   end
 
-  def test_webhook_mergeable_pr_test
-    test_pr_worker = create_test_pr("thumbot/prtester")
-
-    assert test_pr_worker.comments.length == 0
-    assert test_pr_worker.review_count == 0
-    assert test_pr_worker.bot_comments.length == 0
-
-    new_pr_webhook_payload = {
-        'repository' => {'full_name' => test_pr_worker.repo},
-        'number' => test_pr_worker.pr.number,
-        'pull_request' => {'number' => test_pr_worker.pr.number, 'body' => test_pr_worker.pr.body}
-    }
-
-    post '/webhook', new_pr_webhook_payload.to_json
-
-    assert_true last_response.body.include?("OK"), last_response.body
-
-    assert_true test_pr_worker.open?
-    assert test_pr_worker.review_count == 0
-
-    assert test_pr_worker.comments.length == 1
-    assert test_pr_worker.bot_comments.length == 1
-    assert_true test_pr_worker.open?
 
-    assert test_pr_worker.comments.first['body'] =~ /Build Status/
+  def test_can_access_repo
 
-    create_test_code_reviews(test_pr_worker.repo, test_pr_worker.pr.number)
+    cassette(:simple_repo) do
+      repository = Octokit.repo 'thumbot/prtester'
+      assert_not_nil repository
+      assert_equal 'prtester', repository.name
+    end
 
-    assert test_pr_worker.review_count >= 2
+  end
 
-    new_comment_payload = {
-        'repository' => {'full_name' => test_pr_worker.repo},
-        'issue' => {'number' => test_pr_worker.pr.number,
-                    'pull_request' => {}
-        },
-        'comment' => {'body' => "looks good"}
-    }
+  def test_can_get_pr
 
-    assert payload_type(new_comment_payload) == :new_comment, payload_type(new_comment_payload).to_s
+    cassette(:get_basic_pr) do
+      prw = Thumbs::PullRequestWorker.new(:repo => TESTREPO, :pr => TESTPR)
+      assert !prw.nil?
+      assert prw.kind_of?(Thumbs::PullRequestWorker)
+      assert_equal "thumbot/prtester", prw.pr.base.repo.full_name
+      assert_equal "thumbot/prtester", prw.repo
+      assert_equal TESTPR, prw.pr.number
+    end
 
-    post "/webhook", new_comment_payload.to_json
+  end
 
-    assert last_response.body.include?("OK"), last_response.body
+  def test_can_get_status
+    cassette(:get_ratelimit) do
+      get '/status'
+      assert last_response.body.include?("<pre>"), last_response.body
+      last_response_yaml=last_response.body.gsub!(/(\<pre\>|\<\/pre\>)/, '')
+      status_hash=YAML.load(last_response_yaml)
+      assert status_hash.key?(:status), status_hash.inspect
+      assert status_hash.key?(:version)
+      assert status_hash.key?(:rate_limit)
+    end
+  end
+  def test_new_push_hook
 
-    assert_false test_pr_worker.open?
-    test_pr_worker.close
+  end
+  def test_new_comment_hook
 
   end
+  def test_new_pr_hook
+    cassette(:load_pr, :record => :new_episodes) do
+      prw = Thumbs::PullRequestWorker.new(:repo => 'thumbot/prtester', :pr => TESTPR)
```

Comment:
In the future, it might be a good idea to have a config file with the test repos and things defined there rather than hard coded. For our purposes, hard coding is fine, I'm just thinking of when we release this to the public as a full blown open source project.